lib/redix/pubsub/connection.ex

```diff
@@ -230,6 +230,29 @@ defmodule Redix.PubSub.Connection do
     end
   end
 
+  def connected({:call, from}, {:ping, _pid}, data) do
+    random_string = :crypto.strong_rand_bytes(12) |> Base.encode64()
+
+    {:ok, task_pid} =
+      Task.start(fn ->
+        res =
+          receive do
+            {:pong, ^random_string} -> :ok
+          after
+            5000 -> :error
+          end
+
+        :ok = :gen_statem.reply(from, res)
+      end)
```

Comment:
We won't need the task here if we go with a queue of pending clients.

Reply:
Thank you for the feedback.

Would the caller get an `:error` after 5000 ms without the task?

Reply:
The flow to implement here is using gen_statem timeouts and processe aliases, see

> Timeout specifies how long to wait for a response. If no response is received within the specified time, this function returns timeout. Assuming that the server executes on a node supporting aliases (introduced in OTP 24) the request will also be abandoned. That is, no response will be received after a time-out. Otherwise, a stray response might be received at a later time.

from the `gen_statem` docs. Basically you don't care about timeouts, you just reply to every pong and BEAM throws away the responses for procs that timed out already